### ISSUE TITLE
Empty trackable page if GC-website is in english

### ISSUE THREAD

**Issue by murggel** [cgeo/cgeo#13551]
### Describe your problem!

When loading the trackable page (e.g. TB9XHDG) an empty page is shown - only the name in the header is shown.

### How to reproduce?

Switch language at geocaching.com to english

Open the trackable page for a trackable in c:geo

### Actual result after these steps?

GC-Language is english:
![grafik](https://user-images.githubusercontent.com/70113341/197299121-a0279d47-5105-4e78-bae4-e47f8cb22d66.png)

GC-Language is german:
![grafik](https://user-images.githubusercontent.com/70113341/197299538-f490bea5-778e-4495-bef6-76b011f4185a.png)

### Reproducible

Yes

### c:geo Version

2022.10.22

### System information

c:geo version: 2022.10.22-b8e4962 developer build

Device:
-------
- Device type: sdk_gphone_x86 (sdk_gphone_x86, google)
- Available processors: 2
- Android version: 11
- Android build: RSR1.201013.001
- Screen resolution: 1080x2088px (392x735dp)
- Pixel density: 2.75
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 22.41.13 (150700-480714934)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:94
- Set language: en
- System date format: M/d/yy
- Time zone: GMT
- Debug mode active: no
- Last backup: Oct 17, 9:44 PM
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: false / threads: 1
- Map: Germany (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:Cgeo/maps/germany.map
  - Atts: Map data (c) OpenStreetMap contributors
  - Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: None ([:inconclusive=false:advanced=false])
- OFFLINE: None ([:inconclusive=false:advanced=true])

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / BASIC
- Geocaching.com date format: dd.MM.yyyy
- Geocaching.com website language: de-DE
- Last login error on geocaching.com: Failed to log in to geocaching.com as c:geo because of wrong username/password (2022-10-21 10-53)
- Last successful login on geocaching.com: 2022-10-21 22-34
- Routing: internal / BRouter installed: false
- Installed c:geo plugins:  none

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (1.7 GB free) v3 internal isDir(8 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (1.7 GB free) v3 internal isDir(8 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (1.7 GB free) v3 external non-removable isDir(5 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (1.7 GB free) v3 internal isDir(0 entries)
- Public Folders: #12
  - BASE: /Cgeo (User-Defined)[/Cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo, Av:true, files:>=4, dirs:>=9, totalFileSize:>=2.5 GB, free space: 1.7 GB, files on device: 393216)
  - OFFLINE_MAPS: /Cgeo/maps (Default)[/Cgeo/maps[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/maps]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Fmaps, Av:true, files:2, dirs:2, totalFileSize:2.5 GB, free space: 1.7 GB, files on device: 393216)
  - OFFLINE_MAP_THEMES: /Cgeo/maps/_themes (Default)[/Cgeo/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/maps/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Fmaps%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 1.7 GB, files on device: 393216)
  - OFFLINE_MAP_SHADING: /Cgeo/maps/_hgt (Default)[/Cgeo/maps/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/maps/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Fmaps%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 1.7 GB, files on device: 393216)
  - LOGFILES: /Cgeo/logfiles (Default)[/Cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Flogfiles, Av:true, files:1, dirs:0, totalFileSize:522.1 KB, free space: 1.7 GB, files on device: 393216)
  - GPX: /Cgeo/gpx (Default)[/Cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Fgpx, Av:true, files:1, dirs:0, totalFileSize:40.4 KB, free space: 1.7 GB, files on device: 393216)
  - BACKUP: /Cgeo/backup (Default)[/Cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Fbackup, Av:true, files:6, dirs:4, totalFileSize:766.9 KB, free space: 1.7 GB, files on device: 393216)
  - FIELD_NOTES: /Cgeo/field-notes (Default)[/Cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 1.7 GB, files on device: 393216)
  - SPOILER_IMAGES: /Cgeo/GeocachePhotos (Default)[/Cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 1.7 GB, files on device: 393216)
  - ROUTING_BASE: /Cgeo/routing (Default)[/Cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:84.0 KB, free space: 1.7 GB, files on device: 393216)
  - ROUTING_TILES: /Cgeo/routing/segments4 (Default)[/Cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3ACgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ACgeo/document/primary%3ACgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 1.7 GB, files on device: 393216)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:>=5, dirs:>=29, totalFileSize:>=0 B, free space: 1.7 GB, files on device: -1)
- Map render theme path: 
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary%3ACgeo (Jul 1, 1:21 PM):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:700.0 KB) on system internal storage

--- End of system information ---

### Additional Information
[logcat-2022-10-22-00-01-45-2.txt](https://github.com/cgeo/cgeo/files/9843865/logcat-2022-10-22-00-01-45-2.txt)

**Comment by Lineflyer** [comment]
That could match a support ticket I received recently for current release version.
I was unable to reproduce but my gc.om language is german. Will test english as well.

@murggel 
Does it affect all trackables or only specific ones or all?

**Comment by Lineflyer** [comment]
Addon:
Cannot reproduce with current nightly.
GC.com set to english, the above mentioned TB still shows up fine.

**Comment by murggel** [comment]
A second trackable was affected - tested with my account (PM) an with that of my daughter (Basic)

Can reproduce it on my local machine in debugger as well - but did not find time to investigate yet...

**Comment by fm-sys** [comment]
@murggel can you share a logfile?

**Comment by murggel** [comment]
> @murggel can you share a logfile?

I added it to the description

**Comment by murggel** [comment]
> @murggel can you share a logfile? 

Let me know, if you need one with the successful trackable

**Comment by fm-sys** [comment]
Might be related:

```
10-21 23:13:22.121 21804 21804 E cgeo    : java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
10-21 23:13:22.121 21804 21804 E cgeo    : 	at java.util.regex.Matcher.reset(Matcher.java:256)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at java.util.regex.Matcher.<init>(Matcher.java:167)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at java.util.regex.Pattern.matcher(Pattern.java:1027)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at cgeo.geocaching.utils.ImageUtils.forEachImageUrlInHtml(ImageUtils.java:483)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at cgeo.geocaching.utils.ImageUtils.addImagesFromHtml(ImageUtils.java:460)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at cgeo.geocaching.models.Trackable.getImages(Trackable.java:411)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at cgeo.geocaching.TrackableActivity.getOrderedPages(TrackableActivity.java:473)
10-21 23:13:22.121 21804 21804 E cgeo    : 	at cgeo.geocaching.TrackableActivity.displayTrackable(TrackableActivity.java:336)
 
```

**Comment by eddiemuc** [comment]
> Might be related:
> 
> ```
> 10-21 23:13:22.121 21804 21804 E cgeo    : java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
> ...
> ```

Dont know if this NPE is related to the issue at hand, but it is definitely a bug. Thus I fixed it with PR #13554

**Comment by murggel** [comment]
I installed current nightly (with the fix of @eddiemuc ) - seems alright now

So I would close here and reopen if it occurs again
